src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/FilesystemSupplementedRepositoryImpl.java

```diff
@@ -238,28 +248,28 @@ private Type writeFilesToDisk(Path baseDirectory, Type objectToWrite) {
 				Path source = (Path) ReflectionUtils.getField(field, objectToWrite);
 				Path target = sequenceFileDirWithRevision.resolve(source.getFileName());
 				logger.debug("Target is [" + target.toString() + "]");
-				try {
-					if (!Files.exists(sequenceFileDir)) {
-						Files.createDirectory(sequenceFileDir);
-						logger.trace("Created directory: [" + sequenceFileDir.toString() + "]");
-					}
 
-					if (!Files.exists(sequenceFileDirWithRevision)) {
-						Files.createDirectory(sequenceFileDirWithRevision);
-						logger.trace("Created directory: [" + sequenceFileDirWithRevision.toString() + "]");
-					}
+				if(iridaFileStorageService.storageTypeIsLocal()) {
```

Comment:
As much as we can avoid this `if` the better.  If the repo level can just call the filestorageservice to do its directory creation and such that would be better.  Keep the responsibility of managing files to that level if we can.

Reply:
Updated in e9d5d85b90d82ec70a6779780f6b0d0fd138cd82